python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion_with_data.py

```diff
@@ -0,0 +1,369 @@
+
+        Arguments:
+            messages {List[Tuple[str,str]]} -- The messages to use for the chat completion.
+            settings {ChatRequestSettings} -- The settings to use for the chat completion request.
+            logger {Optional[Logger]} -- The logger instance to use. (Optional)
+
+        Returns:
+            Union[Tuple[str, str], List[Tuple[str,str]]] -- The completion result(s)
+            in the format (assistant_message, tool_message). The tool message contains additional
+            information about the data source including citations.
+        """
+        response = await self._send_request(
+            messages=messages, request_settings=settings, stream=False
+        )
+
+        if len(response.choices) == 1:
+            return self._parse_message(response.choices[0].message)
+        else:
+            return [self._parse_message(choice.message) for choice in response.choices]
+
+    def _parse_message(
+        self, message: ChatCompletionMessage, functions: bool = False
+    ) -> Tuple[str, str]:
+        """Parses the message from the response, returns a tuple of (assistant response, tool response)."""
+        assistant_content = message.content
+        tool_content = ""
+        if message.model_extra and "context" in message.model_extra:
+            for m in message.model_extra["context"].get("messages", []):
+                if m["role"] == "tool":
+                    tool_content = m.get("content", "")
+                    break
+
+        if functions:
+            function_call = (
+                message.function_call if hasattr(message, "function_call") else None
+            )
+            if function_call:
+                function_call = FunctionCall(
+                    name=function_call.name,
+                    arguments=function_call.arguments,
+                )
+            return (assistant_content, tool_content, function_call)
+
+        return (assistant_content, tool_content)
+
+    async def complete_chat_stream_async(
+        self,
+        messages: List[Dict[str, str]],
+        settings: "ChatRequestSettings",
+        logger: Optional[Logger] = None,
+    ) -> AsyncGenerator[Union[str, List[str]], None]:
+        """Executes a chat completion request and returns the result.
+
+        Arguments:
+            messages {List[Tuple[str,str]]} -- The messages to use for the chat completion.
+            settings {ChatRequestSettings} -- The settings to use for the chat completion request.
+            logger {Optional[Logger]} -- The logger instance to use. (Optional)
+
+        Returns:
+            Union[str, List[str]] -- The completion result(s).
+        """
+        response = await self._send_request(
+            messages=messages, request_settings=settings, stream=True
+        )
+
+        return AzureChatWithDataStreamResponse(response, settings)
+
+    @staticmethod
+    def _parse_choices(choice) -> Tuple[str, int]:
```

Comment:
This method looks exactly like the original implementation in `open_ai_chat_completion_base.OpenAIChatCompletionBase` which this class inherited. Do we need to reimplement this here?

Reply:
Refactored to use the base class.